lib/ansible/modules/net_tools/ldap/_ldap_attr.py

```diff
@@ -255,11 +257,28 @@ def main():
         module.fail_json(msg=missing_required_lib('python-ldap'),
                          exception=LDAP_IMP_ERR)
 
-    # Update module parameters with user's parameters if defined
-    if 'params' in module.params and isinstance(module.params['params'], dict):
+    # For Ansible-2.9.x and below, allow the params module parameter with a warning
+    if LooseVersion(module.ansible_version) < LooseVersion('2.10'):
+        if module.params['params']:
+            module.deprecate("The `params` option to ldap_attr will be removed in Ansible 2.10"
+                             " since it circumvents Ansible's option handling", version='2.10')
+
+        # However, the bind_pw parameter contains a password so it **must** go through the normal
+        # argument parsing even though removing it breaks backwards compat.
+        if 'bind_pw' in module.params['params']:
+            module.fail_json("Using `bind_pw` with the `params` option has been disallowed since"
```

Comment:
`fail_json` requires a `msg` keyword argument. Needs to be updated for the other places fail_json was added too.

Reply:
Doh.... I'll fix that here and also open an issue to change that behaviour in fail_json.  It's a mandatory parameter so there's no reason it shouldn't be accessible as a positional arg too.

Reply:
Fix pushed